### PR TITLE
Issue #21576 rename dpathextractor fieldpointer

### PR DESCRIPTION
**Pull request by maxi297**
## What
Addresses #21576 

## How
* Rename DpathExtractor.field_pointer to field_path in declarative_component_schema and regenerate files
* Update DpathExtractor.field_pointer to field_path (not the model but the actual component we use)
* Fixes sources (some source-monday and source-pocket were re-implementing DpathExtractor)
* Update Connector Builder Server
* Migration (see https://github.com/airbytehq/airbyte/pull/21997)

## Recommended reading order
1. `airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml`
2. `airbyte-cdk/python`
3. `airbyte-integrations`
4. `airbyte-connector-builder-server`

## 🚨 User Impact 🚨
This is a breaking change that is part of the Low-Code CDK to Beta

### CODE REVIEW

File: airbyte-connector-builder-server/connector_builder/generated/models/dpath_extractor.py [hunk 1]
```diff
@@ -19,13 +19,13 @@ class DpathExtractor(BaseModel):
...
     DpathExtractor - a model defined in OpenAPI
 
-        field_pointer: The field_pointer of this DpathExtractor.
+        field_path: The field_path of this DpathExtractor.
         config: The config of this DpathExtractor.
         decoder: The decoder of this DpathExtractor [Optional].
     """
 
-    field_pointer: List[AnyOfInterpolatedStringstring] = Field(alias="field_pointer")
+    field_path: List[AnyOfInterpolatedStringstring] = Field(alias="field_path")
```
Comment (maxi297) [inline airbyte-connector-builder-server/connector_builder/generated/models/dpath_extractor.py:27]: This file was autogenerated in December 2022. However, I can't find what is the process to re-generate this. I see [this](https://github.com/airbytehq/airbyte/blob/ca8cdc4/airbyte-connector-builder-server/src/main/openapi/openapi.yaml#L222) has been commented out but I don’t see any alternative solution provided. Also, the file used seems to be the previous json file and not `declarative_component_schema.yaml`. Based on the commit/issues associated, it seems there was a lot of struggle with code generation. Therefore, I have a couple of questions:
* Do we need code generation for the manifest in the Connector Server Builder?
* If so, should we clarify the state of the code generation in our README.md?
* If not, should we clean `openapi.yaml` file?

File: airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py [hunk 2]
```diff
@@ -83,12 +83,12 @@ class NestedDpathExtractor(DpathExtractor):
 
     def extract_records(self, response: requests.Response) -> List[Record]:
         response_body = self.decoder.decode(response)
-        if len(self.field_pointer) == 0:
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py:86]: `NestedDpathExtractor` seems to be something that would be interesting in the airbyte-cdk per se. If we agree, I could create a Continuous Improvement ticket to migrate it (or make DpathExtractor support the same functionality)

Note: To be clear, I haven't checked much the difference and I'm not clear on what the gap between `DpathExtractor` and `NestedDpathExtractor` is

File: airbyte-integrations/connectors/source-facebook-pages/unit_tests/test_facebook_authenticator.py [hunk 3]
```diff
@@ -19,13 +19,13 @@ def test_facebook_url_params(req_mock):
         "access_token": "initial_token",
         "page_id": "pageID"
     }
-    options = config
+    parameters = config
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-pages/unit_tests/test_facebook_authenticator.py:22]: The changes from `options` to `parameters` was part of https://github.com/airbytehq/airbyte/issues/19408 but some sources were not updated. I updated the sources that were impacted by my changes

File: airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py (hunk 2 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py:86]: I think we can deprecate the nested one: https://github.com/airbytehq/airbyte/pull/21690

File: airbyte-cdk/python/bin/migrate_field_point_to_field_path.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (sherifnada) [inline airbyte-cdk/python/bin/migrate_field_point_to_field_path.sh:1]: request: let's make sure we remove these scripts after beta migration is done

**Review by sherifnada — APPROVED** [review]

**Review by clnoll — APPROVED** [review]
LGTM @maxi297!

File: airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py (hunk 2 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-facebook-pages/source_facebook_pages/components.py:86]: Confirmed with artem1205 that GL will remove this eventually

**Pull request merged**